### PR TITLE
Us 02

### PR DESCRIPTION
**Pull request by MaaaChu**

### CODE REVIEW

File: klattice/src/app/components/role-modal/role-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export class RoleModalComponent {
+
+  title = "Test title";
+  jobTitle = "Software Engineer";
+
+  constructor() {}
+
+  // open(content) {
+  //   this.modalService.open(content, {size: 'lg', ariaLabelledBy: 'modal-basic-title'})
+  // }
```
Comment (jamescrone) [inline klattice/src/app/components/role-modal/role-modal.component.ts]: same comment from above applies here

File: klattice/src/app/components/role-modal/role-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      </div>
+    </form>
+  </div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-outline-dark" (click)="modal.close('Save click')">Save</button>
+  </div>
+</ng-template>
+
+<!-- <button class="btn btn-lg btn-outline-primary" (click)="open(content)">Launch demo modal</button> -->
+
```
Comment (jamescrone) [inline klattice/src/app/components/role-modal/role-modal.component.html]: Commented out line may be confusing to others pulling from master, is this line of code needed or can it be removed?

File: SQLstructure.sql [hunk 3]
```diff
@@ -4,13 +4,80 @@ USE klattice;
 
 CREATE TABLE role (
 	role_id INT unsigned auto_increment,
-    role_name varchar(60),
+    role_name varchar(100) not null,
+    role_summary varchar(400),
+    role_sum_link varchar(250),
```
Comment (matthewdoreilly) [inline SQLstructure.sql]: Is `role_sum_link` a SharePoint URL to the job spec? From your inserts, it's always set to null.

File: api/db.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const db = mysql.createConnection({
 
 exports.getRoles = function (callback) {
     db.query(
-        "SELECT role_id, role_name FROM role;",
+        "SELECT role_id, role_name, role_summary, role_sum_link FROM role;",
```
Comment (matthewdoreilly) [inline api/db.js]: You probably don't need to return `role_id` here.

File: klattice/src/app/app.component.html [hunk 5]
```diff
@@ -24,4 +24,26 @@ <h1>
...
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <div class="modal-body">
+    <form>
+      <div class="form-group">
+        <h5 class="jobSpec">Job Specification</h5>
+        <p>{{roleSummary}}</p>
+        <br>
+        <a [href]=roleLink>Sharepoint Link</a>
```
Comment (matthewdoreilly) [inline klattice/src/app/app.component.html]: Attribute name should be just `href` here (without the square brackets). Strange it still displays as a link on the frontend though

File: klattice/src/app/components/role-modal/role-modal.component.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        <p>Hello, {{title}}!</p>
+      </div>
+    </form>
+  </div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-outline-dark" (click)="modal.close('Save click')">Save</button>
+  </div>
+</ng-template>
+
+<!-- <button class="btn btn-lg btn-outline-primary" (click)="open(content)">Launch demo modal</button> -->
```
Comment (matthewdoreilly) [inline klattice/src/app/components/role-modal/role-modal.component.html]: Remove commented code and any unnecessary whitespace.

File: klattice/src/app/components/role-modal/role-modal.component.ts (hunk 1 above)
Comment (Broom94) [reply klattice/src/app/components/role-modal/role-modal.component.ts]: done

File: api/db.js (hunk 4 above)
Comment (matthewdoreilly) [reply api/db.js]: Talked to @Mark-Spiers about this. It's ok to select the `role_id` here to use in your frontend code, but I'd not show it on the actual HTML view, as I don't think it's really useful for an end user to see it.

File: klattice/src/app/components/role-modal/role-modal.component.html (hunk 6 above)
Comment (Broom94) [reply klattice/src/app/components/role-modal/role-modal.component.html]: done

File: klattice/src/app/app.component.html (hunk 5 above)
Comment (Broom94) [reply klattice/src/app/app.component.html]: done

File: api/db.js (hunk 4 above)
Comment (Broom94) [reply api/db.js]: spoken to mark, done

File: SQLstructure.sql (hunk 3 above)
Comment (Broom94) [reply SQLstructure.sql]: it's set in some just to test
